spectrum_utils/plot.py

```diff
@@ -286,3 +430,83 @@ def mirror(
     )
 
     return ax
+
+
+def facet(
+    spec_top: MsmsSpectrum,
+    spec_mass_errors: Optional[MsmsSpectrum] = None,
+    spec_bottom: Optional[MsmsSpectrum] = None,
+    spectrum_kws: Optional[Mapping[str, Any]] = None,
+    mass_errors_kws: Optional[Mapping[str, Any]] = None,
+    height: Optional[float] = None,
+    aspect: Optional[float] = None,
+) -> plt.Figure:
+    """
+    Plot a spectrum, and optionally mass errors, and a mirror spectrum.
+
+    Parameters
+    ----------
+    spec_top : MsmsSpectrum
+        The spectrum to be plotted on the top.
+    spec_mass_errors : Optional[MsmsSpectrum], optional
+        The spectrum for which mass errors are to be plotted in the middle.
+    spec_bottom : Optional[MsmsSpectrum], optional
+        The spectrum to be plotted on the bottom.
+    spectrum_kws : Optional[Mapping[str, Any]], optional
+        Keyword arguments for `plot.spectrum` for the top and bottom spectra.
+    mass_errors_kws : Optional[Mapping[str, Any]], optional
+        Keyword arguments for `plot.mass_errors`.
+    height : Optional[float], optional
+        The height of the figure in inches.
+    aspect : Optional[float], optional
+        The width of the figure as an aspect ratio of the height.
+
+    Returns
+    -------
+    plt.Figure
+        The matplotlib Figure instance on which the spectra and mass errors
+        are plotted.
+    """
+
+    n_rows = 1 + (spec_mass_errors is not None) + (spec_bottom is not None)
+    height_ratios = [1]
+    if spec_mass_errors is not None:
+        height_ratios.append(0.5)
+    if spec_bottom is not None:
+        height_ratios.append(1)
+
+    fig, axes = plt.subplots(
+        *(n_rows, 1),
+        figsize=(10, 8 if spec_bottom is not None else 5),
```

Comment:
**question:** Can the figure size be directly set here instead of with `set_size_inches` below?

Reply:
Yes. I think I used `set_size_inches` at some point while tweaking the function's order of operations, but no reason to keep it there.